File: strengthScaler_AllMuscles.m
function modelOutputPath = strengthScaler_AllMuscles(modelPath,muscles_num,n,resultsFolder,addName)

%%muscles_num --> nMUS=OSmodel.getMuscles() and then nMUS.getSize()

% Import OpenSim Libraries
import org.opensim.modeling.*      

if nargin < 4
    resultsFolder = 'Results';
end

[pathName, modelName, ext] = fileparts(modelPath); 

% create instance of the model 
myModel = Model(modelPath);
myModel.initSystem();

% make the scaling factor a decimal (if 100% then 1, if 50% then 0.5
scalingFactor = n/100;

% multiply muscles -found in muscleList- in the model by scalingFactor
for ii = 0 : muscles_num-1
    b = myModel.getMuscles.get( ii ).getMaxIsometricForce ;
    myModel.getMuscles.get( ii ).setMaxIsometricForce( b*scalingFactor ) ;
end

% Print the model out to the results file, ready to be used by the
% cmcTool
modelOutputPath = fullfile(resultsFolder, ['myModel_' addName '_' num2str(round(n)) '.osim']) ;
myModel.print(modelOutputPath);

display('model printed')

end 

File: strengthScaler_AllMuscles.py
import opensim as osim
import os


def strengthScaler_AllMuscles(modelPath, muscles_num, n, resultsFolder='Results', addName=''):
    """Scale the max isometric force of all muscles in an OpenSim model.

    Parameters
    ----------
    modelPath : str
        Full path to the OpenSim model file.
    muscles_num : int
        Number of muscles in the model. Obtain via
        OSmodel.getMuscles().getSize().
    n : float
        Scaling percentage (e.g. 100 = no change, 50 = half strength).
    resultsFolder : str, optional
        Folder where the scaled model will be saved. Default is 'Results'.
    addName : str, optional
        Additional name tag for the output model file.

    Returns
    -------
    modelOutputPath : str
        Full path to the printed scaled model.
    """

    # create instance of the model
    myModel = osim.Model(modelPath)
    myModel.initSystem()

    # make the scaling factor a decimal (if 100% then 1, if 50% then 0.5)
    scalingFactor = n / 100

    # multiply all muscles in the model by scalingFactor
    for ii in range(muscles_num):
        b = myModel.getMuscles().get(ii).getMaxIsometricForce()
        myModel.getMuscles().get(ii).setMaxIsometricForce(b * scalingFactor)

    # Print the model out to the results file, ready to be used by the
    # cmcTool
    modelOutputPath = os.path.join(resultsFolder,
                                   'myModel_' + addName + '_' + str(round(n)) + '.osim')
    myModel.printToXML(modelOutputPath)

    print('model printed')

    return modelOutputPath
